lensless/recon/trainable_recon.py

```diff
@@ -58,29 +66,87 @@ def __init__(self, psf, dtype=None, n_iter=1, **kwargs):
                 Data type to use for optimization.
             n_iter : int
                 Number of iterations for unrolled algorithm.
+            post_process : :py:class:`function` or :py:class:`~torch.nn.Module`, optional
+                If :py:class:`function` : Function to apply to the image estimate after the whole ADMM algorithm. Its input most be (image to process, noise_level), where noise_level is a learnable parameter. If it include aditional learnable parameters, they will not be added to the parameter list of the algorithm. To allow for traning, the function must be autograd compatible.
+                If :py:class:`~torch.nn.Module` : A DruNet like network. The network will be included as a submodule of the algorithm and its parameters will be added to the parameter list of the algorithm. If this isn't intended behavior, set requires_grad=False.
```

Comment:
does it have to be a DruNet-like architecture? or just any torch module

Reply:
or you can say UNet-like architecture to be slightly more general

Reply:
Any unet will not work. We need one with 4 channel as input and 3 as output. I will try to be more specific.